File: JanperCab/3-Application/Dtos/CustomerCategory/CustomerCategoryAccountDto.cs
using _1_Domain.Enum;

namespace _3_Application.Dtos.CustomerCategory
{
    public class CustomerCategoryAccountDto : CustomerCategoryDto
    {
        public int DurationInDays { get; set; }

        public CustomerCategoryAccountDto()
        {
            CategoryType = CustomerCategoryType.Account;
        }
    }
}

File: JanperCab/5-JanperCab/ClientApp/src/app/_models/customer-category/CustomerCategoryAccountDto.ts
import { CustomerCategoryType } from 'src/app/_enums/CustomerCategoryType';
import { CustomerCategoryDto } from './CustomerCategoryDto';

export class CustomerCategoryAccountDto extends CustomerCategoryDto {
  durationInDays: number;

  constructor() {
    super();
    this.$type =
      '_3_Application.Dtos.CustomerCategory.CustomerCategoryAccountDto, 3-Application';
    this.categoryType = CustomerCategoryType.Account;
  }

  get description(): string {
    return `Account - ${this.durationInDays} days`;
  }
}
